src/mod_auth_cas.c

```diff
@@ -2217,29 +2229,35 @@ int cas_authenticate(request_rec *r)
 			if(d->CASAuthNHeader != NULL)
 				apr_table_set(r->headers_in, d->CASAuthNHeader, remoteUser);
 
-			if(parametersRemoved == TRUE) {
-				if(ssl == TRUE && port != 443)
-					printPort = TRUE;
-				else if(port != 80)
-					printPort = TRUE;
-
-				if(c->CASRootProxiedAs.is_initialized) {
+			if(d->CASDisableRedirectAfterValidation == NULL) {
```

Comment:
Why not:

```
                       if(d->CASDisableRedirectAfterValidation) {
                               return OK;
                       }
```

and keep everything in the "if(parametersRemoved == TRUE)" unchanged? The patch and code are much more readable like that.